### ISSUE TITLE
is it possible to tweak `multifasta_query` output format so that cDBG node ids are annotated with record in a csv file?

### ISSUE THREAD

**Issue by taylorreiter** [spacegraphcats/spacegraphcats#420]
```
python -m spacegraphcats run twofoo multifasta_query
```

Runs sgc for this conf file:
```
catlas_base: twofoo
input_sequences:
- twofoo.fq.gz
ksize: 31
radius: 1
search:
- data/2.fa.gz
- data/47.fa.gz
- data/63.fa.gz
shadow_ratio_maxsize: 1000

hashval_ksize: 51
hashval_queries: data/twofoo-k51-hashval-queries.txt

multifasta_reference:
- data/twofoo-genes.fa.gz
multifasta_scaled: 1000
multifasta_query_sig: data/63-os223.sig
```

Producing:
```
twofoo_k31_r1_multifasta:
    hashval.pickle
    multifasta.cdbg_by_record.csv
    multifasta.pickle
    query-results.csv
    record0-nbhd.cdbg_ids.txt.gz
    record1-nbhd.cdbg_ids.txt.gz
    record2-nbhd.cdbg_ids.txt.gz
    record3-nbhd.cdbg_ids.txt.gz
    record4-nbhd.cdbg_ids.txt.gz
    record5-nbhd.cdbg_ids.txt.gz
    record6-nbhd.cdbg_ids.txt.gz
```

`twofoo-genes.fa.gz` contains sequences with the following headers:
```
>lcl|NC_011663.1_cds_WP_006083826.1_1 [gene=dnaA] [locus_tag=SBAL223_RS00005] [protein=chromosomal replication initiator protein DnaA] [protein_id=WP_006083826.1] [location=77..1465] [gbkey=CDS]
>lcl|NC_011663.1_cds_WP_006083825.1_2 [gene=dnaN] [locus_tag=SBAL223_RS00010] [protein=DNA polymerase III subunit beta] [protein_id=WP_006083825.1] [location=1485..2585] [gbkey=CDS]
>lcl|NC_011663.1_cds_WP_006083824.1_3 [gene=recF] [locus_tag=SBAL223_RS00015] [protein=DNA replication/repair protein RecF] [protein_id=WP_006083824.1] [location=2787..3869] [gbkey=CDS]
>lcl|NC_011663.1_cds_WP_006084772.1_4 [gene=gyrB] [locus_tag=SBAL223_RS00020] [protein=DNA topoisomerase (ATP-hydrolyzing) subunit B] [protein_id=WP_006084772.1] [location=3886..6303] [gbkey=CDS]
>lcl|NC_011663.1_cds_WP_012586329.1_5 [locus_tag=SBAL223_RS00025] [protein=glutathione S-transferase family protein] [protein_id=WP_012586329.1] [location=complement(6418..7080)] [gbkey=CDS]
>lcl|NC_011663.1_cds_WP_012586330.1_6 [locus_tag=SBAL223_RS00030] [protein=cytochrome c] [protein_id=WP_012586330.1] [location=7321..7623] [gbkey=CDS]
>lcl|NC_011663.1_cds_WP_012586331.1_7 [locus_tag=SBAL223_RS00035] [protein=sulfite oxidase] [protein_id=WP_012586331.1] [location=7625..8851] [gbkey=CDS]
```

Which are referred to in the `multifasta` output dir as `record0`, `record1`, ..., `record6`. 

`record0-nbhd.cdbg_ids.txt.gz` contains the cdbg ids that match that are in the nbhd of the annotation: 
```
142981
38533
56974
6304
142114
184614
67504
191540
31798
66359
8775
1361
181333
168544
43107
146534
46439
179304
11247
93942
89082
```

I can create this file myself in R etc via postprocessing, but it would be really awesome if `multifasta` output would include a file called something like `multifasta.cdbg_results.csv`:
(I abbreviated gene name for clarity. It should not be abbreviated in the real output)

```
filename,record_name,record_number,catlas_base,cdbg_id
twofoo-genes.fa.gz, lcl|NC_011663.1_cds_WP_006083826.1_1 [gene=dnaA], record0, twofoo, 142981
twofoo-genes.fa.gz, lcl|NC_011663.1_cds_WP_006083826.1_1 [gene=dnaA], record0, twofoo, 38533
twofoo-genes.fa.gz, lcl|NC_011663.1_cds_WP_006083826.1_1 [gene=dnaA], record0, twofoo, 56974
```

@ctb

**Comment by taylorreiter** [comment]
could be added to `spacegraphcats/search/extract_cdbg_by_multifasta.py`:

```
output2 = "multifasta.cdbg_results.csv"
with open(output2, "wt") as fp:
    w = csv.writer(fp)
    w.writerow(
        [
            "filename",
            "record_name",
            "record_number",
            "catlas_base",
            "cdbg_id",
        ]
    )
    
    filenum = 0
    for (filename, record_name), cdbg_ids in records_to_cdbg.items():
        if not cdbg_ids:
            continue

        # write out cdbg id to record mapping, one per row
        record_number = f"record{filenum}"
        for cdbg_id in cdbg_ids:
            cdbg_id = str(cdbg_id)
            w.writerow(
                [
                    filename,
                    record_name,
                    record_number,
                    catlas_base,
                    cdbg_id,
                ]
            )

        filenum += 1

```

**Comment by taylorreiter** [comment]
Closed by #422
